### ISSUE TITLE
Handle constants in attribute 0 on Safari WebGL1

### ISSUE THREAD

**Issue by ibgreen** [uber/luma.gl#609]
#### Actual Result
- constants in attribute 0 do not work on Safari, Desktop OpenGL spec does not allow app to disable attribute 0, and no reliable way to control which attribute gets assigned to location 0.

#### Expected Result
- constants in attribute 0 work.

#### Reproduce Steps
- Website pointcloudlayer

#### Background 
- Proposed solution: If constant is set on attribute 0 on browsers that do not already support it, allocate a buffer and fill it with the constant value, bind it and enable the attribute.

This is likely what e.g. Firefox does behind the scenes (which is why they issue a perf warning when disabling attribute 0).

Note: OpenGL ES does allow 0 to be disabled, so this is only an issue on desktops. Chrome seems to work around it quietly, perhaps also using the suggested trick internally?

#### To Do List 
  - [x] Add label and assign to milestone
  - [x] Coding
  - [x] Test

@1chandu @Pessimistress @jianhuang01

**Comment by Pessimistress** [comment]
If this is the case, how is generic attribute working on Safari in 5.3?

**Comment by ibgreen** [comment]
> If this is the case, how is generic attribute working on Safari in 5.3?

Good question. 

My understanding is simply that the assignment of locations to attributes is non-deterministic in the sense that the program linker/GPU driver decides based on factors we cannot predict (e.g. alignment of various attributes/uniforms/varyings etc). 

Essentially, as long as no attribute (that we assign constants to) gets assigned to attribute 0, things work.

Turning on logging in deck.gl 5.3 `PointLayer` shows that `instanceColors` gets assigned to location 1, but on 6.0 the same attribute gets assigned to location 0. I tried reordering the attributes in the vertex shader source but it didn't make a difference.

Why the difference? Maybe the point cloud shaders have changed in v6.0? If not, a lot of code is injected into the shaders (prologue and modules), maybe that has changed, causing the program linker to make different decisions in v6.0.

Maybe we are using more constant values in v6.0, increasing the risk we hit attribute 0?

Also not all 5.3 layers render for me in Safari (e.g. LineLayer fails). I don't see the attribute 0 issue right now, but I am not convinced we have tested 5.3 rigorously enough on Safari to be able to say that this didn't happen before.
